File: backend-common/src/main/java/cc/jktu/apps/common/dao/entity/BaseEntity.java
package cc.jktu.apps.common.dao.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import lombok.Data;

@Data
public abstract class BaseEntity {

    @TableId(type = IdType.ASSIGN_ID)
    private Long id;
    @TableField(fill = FieldFill.INSERT)
    private Long createTime;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private Long updateTime;

}

File: backend-common/src/main/kotlin/cc/jktu/apps/common/dao/entity/BaseEntity.kt
package cc.jktu.apps.common.dao.entity

import com.baomidou.mybatisplus.annotation.FieldFill
import com.baomidou.mybatisplus.annotation.IdType
import com.baomidou.mybatisplus.annotation.TableField
import com.baomidou.mybatisplus.annotation.TableId
import lombok.Data

@Data
abstract class BaseEntity(
    @TableId(type = IdType.ASSIGN_ID)
    var id: Long? = null,
    @TableField(fill = FieldFill.INSERT)
    var createTime: Long? = null,
    @TableField(fill = FieldFill.INSERT_UPDATE)
    var updateTime: Long? = null
)
